docs/modules/ROOT/pages/architecture/traits.adoc

```diff
@@ -1,6 +1,58 @@
+
+Another important concept related to the trait lifecycle is the **trait profile**. At this time, Camel K is supporting the following profiles:
+
+* Kubernetes
+* Openshift
+* Knative
+
+A profile is useful to identify on which kind of cluster a trait has to run: vanilla `Kubernetes`, `Openshift` or OpenShift/Kubernetes clusters powered by `Knative`.
+
+=== Trait configuration
+
+A Camel K user will provide a trait configuration via CLI (`--trait` or `-t` flag) or setting via `Integration` metadata. Each trait has a unique id and can expose any kind of parameter ie, `kamel run -t [trait-id].[key]=[value]`. The operator will transform that _key_ and _value_ in the related trait variable.
+
+=== Trait interface
+
+In order to understand better the logic behind the trait managment, let's have a look at the `https://github.com/apache/camel-k/blob/main/pkg/trait/trait_types.go#L70[Trait]` interface:
+
+[source,go]
+----
+type Trait interface {
+	Identifiable
+	client.Injectable
+	InjectContext(context.Context)
+	Configure(environment *Environment) (bool, error)
+	Apply(environment *Environment) error
+	InfluencesKit() bool
+	IsPlatformTrait() bool
+	RequiresIntegrationPlatform() bool
+	IsAllowedInProfile(v1.TraitProfile) bool
+	Order() int
+}
+----
+
+Each trait will implement this interface. The most important functions that will be invoked by the xref:architecture/operator.adoc[Operator] are `Configure()` and `Apply()`. Basically, the `Configure()` function will set those inputs aforementioned (each trait has its own). The function is in charge to verify also the correctness of those expected parameters, where it makes sense (ie, a well expected `Kubernetes` resource name).
+
+Once configured, the `Apply()` function will be called along the build or initialization phase in order to do the business logic expected for it. The `environment` variable will give you all the below resources you will need to perform your operation (ie, the `Integration` or any Kubernetes resource attached to it). You can have a deeper look at the `https://github.com/apache/camel-k/blob/main/pkg/trait/trait_types.go#L188[Environment]` struct.
+
+The `Order()` function helps in resolving the order of execution of different traits. As every trait can be expected to be run before or after another trait, or any other controller operation.
+
+The functions `InfluencesKit()`, `IsPlatformTrait()` and `RequiresIntegrationPlatform()` are easy to understand. They are used to determine if a trait has to influece an `IntegrationKit` build/initiatlization, if it's a platform trait (ie, needed by the platform itself) or are requiring the presence of an `IntegrationPlatform`.
```

Comment:
I'd suggest: The `InfluencesKit()`, `IsPlatformTrait()` and `RequiresIntegrationPlatform()` methods ...